doc/atomvm-memory.md

```diff
@@ -0,0 +1,538 @@
+# AtomVM Memory Management
+
+Like most managed execution environments, AtomVM provides automated memory management for compiled Erlang/Elixer applications that run on the platform, allowing developers to focus on the logic of application programs, instead of the minutiae of managing the allocation and disposal of memory in the process heap of the program.
```

Comment:
```suggestion
Like most managed execution environments, AtomVM provides automated memory management for compiled Erlang/Elixir applications that run on the platform, allowing developers to focus on the logic of application programs, instead of the minutiae of managing the allocation and disposal of memory in the process heap of the program.
```